gateway/src/apicast/proxy.lua

```diff
@@ -64,6 +64,11 @@ function _M.new(configuration)
     configuration = assert(configuration, 'missing proxy configuration'),
     cache = cache,
     cache_handler = cache_handler,
+
+    -- Params to send in 3scale backend calls that are not the typical ones
+    -- (credentials, usage, etc.).
+    -- This allows us, for example, to send a referrer.
+    extra_params_backend_authrep = {}
```

Comment:
I opted for this because it's way easier than propagating these extra params to all the methods that need them.
I think this makes the code more readable.

Reply:
I guess it is fine. Proxy object mutates internal state anyway, so it is not really ready to be used by multiple clients.